src/sidebar/components/new-note-btn.js

```diff
@@ -1,25 +1,31 @@
 import { createElement } from 'preact';
 import propTypes from 'prop-types';
 
-import events from '../events';
+import uiConstants from '../ui-constants';
 import useStore from '../store/use-store';
 import { withServices } from '../util/service-context';
 import { applyTheme } from '../util/theme';
 
 import Button from './button';
 
-function NewNoteButton({ $rootScope, settings }) {
-  const store = useStore(store => ({
-    frames: store.frames(),
-  }));
+function NewNoteButton({ settings }) {
+  const frames = useStore(store => store.frames());
+  const isLoggedIn = useStore(store => store.isLoggedIn());
+
+  const createAnnotation = useStore(store => store.createAnnotation);
+  const openSidebarPanel = useStore(store => store.openSidebarPanel);
 
   const onNewNoteBtnClick = function() {
-    const topLevelFrame = store.frames.find(f => !f.id);
+    if (!isLoggedIn) {
+      openSidebarPanel(uiConstants.PANEL_LOGIN_PROMPT);
+      return;
+    }
+    const topLevelFrame = frames.find(f => !f.id);
```

Comment:
There is an easier way to get this now. You can use the `store.mainFrame()` selector, which returns the same value as `topLevelFrame`.

Reply:
Thanks! Fixed.